tower-test/src/lib.rs

```diff
@@ -6,3 +6,21 @@
 
 mod macros;
 pub mod mock;
+
+use futures_util::future::{ready, Ready};
+use std::task::{Context, Poll};
+use tower_service::Service;
+
+/// A type for use in doctests.
+pub struct MyService;
```

Comment:
feature flag this?

Reply:
or I guess doc-tests should use something like service_fn?

Reply:
Ah, to be clear, this is useful in situations like this: https://github.com/tower-rs/tower/pull/330/files#diff-9631c0644350438ce2d92778f80033d3R90
Where you really just want to show that _some_ service is passed in. I guess we could replace it with `service_fn`, but in these examples all that matters is that there is _some_ user-provided service, so the less noise the better.

Reply:
So I _think_ I'd prefer to keep it the way it is. Why would you like to have it under a feature flag?

Reply:
The other option would be to do docs like

```rust
# async fn thing(svc: impl Service<()>) {
svc.call(()).await.unwrap();
# }
```

but im fine either way